src/pac.c

```diff
@@ -109,28 +110,66 @@ pac_may_have_muzzy(pac_t *pac) {
 }
 
 static edata_t *
-pac_alloc_impl(tsdn_t *tsdn, pai_t *self, size_t size, size_t alignment,
-    bool zero, bool *deferred_work_generated) {
-	pac_t *pac = (pac_t *)self;
-
-	*deferred_work_generated = false;
+pac_alloc_real(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks, size_t size,
```

Comment:
What's implied by `real` here?

Reply:
It's doing the actual alloc work. Normally this type of func will be called `_impl`. However it's invoked by both the non-guarded and guarded cases (that are already named `_impl`, aligned with other similar functions), thus why separating a `_real` function out.